src/models/message_status.ts

```diff
@@ -18,22 +18,29 @@ import {
   RetrievedVersionedModel
 } from "../utils/cosmosdb_model_versioned";
 import { wrapWithKind } from "../utils/types";
+import { FiscalCode } from "../../generated/definitions/FiscalCode";
 
 export const MESSAGE_STATUS_COLLECTION_NAME = "message-status";
 export const MESSAGE_STATUS_MODEL_ID_FIELD = "messageId" as const;
 export const MESSAGE_STATUS_MODEL_PK_FIELD = "messageId" as const;
 
 // We cannot intersect with MessageStatus
 // as it is a *strict* interface
-export const MessageStatus = t.interface({
-  messageId: NonEmptyString,
-  status: MessageStatusValue,
-  updatedAt: Timestamp,
-  // eslint-disable-next-line sort-keys
-  isRead: withDefault(t.boolean, false),
-  // eslint-disable-next-line sort-keys
-  isArchived: withDefault(t.boolean, false)
-});
+export const MessageStatus = t.intersection([
+  t.interface({
+    messageId: NonEmptyString,
+    status: MessageStatusValue,
+    updatedAt: Timestamp,
+    // eslint-disable-next-line sort-keys
+    isRead: withDefault(t.boolean, false),
+    // eslint-disable-next-line sort-keys
+    isArchived: withDefault(t.boolean, false)
+  }),
+  // fiscalCode is optional due to retro-compatibility
+  t.partial({
+    fiscalCode: FiscalCode
```

Comment:
We broadly use plain fiscal codes on our models, however we agreed on that we would have built our codebase around hashed values instead. Consider doing so.

Reply:
I agree with you, but we need pain fiscal code here because it's used as partition key for querying `messages` collection.